### PR TITLE
New option to trip a busbar section in the SLD.

### PR DESCRIPTION
**Pull request by EstherDarkish**
Needs to update the OperatingStatus in Powsybl-core and add the BUSBAR_SECTION in the isAllowedIdentifiable function to test properly.
Ask me how to if the next version of powsybl is not used yet.

Edit : This is ready

### CODE REVIEW

**Review by antoinebhs — COMMENTED** [review]
* we should disable the option when the BBS is already tripped (similarly to lines or other equipment) to prevent user tripping twice the same BBS
* add a "switch on" option to revert a trip ?

File: src/components/menus/bus-menu.tsx [hunk 1]
```diff
@@ -80,6 +91,31 @@ export const BusMenu: FunctionComponent<BusMenuProps> = ({
...
+    function startModification() {
+        onClose();
+        if (setModificationInProgress !== undefined) {
+            setModificationInProgress(true);
+        }
+    }
+
+    function handleTrip() {
+        startModification();
+        const equipmentInfos = { id: busId }; // We only need the ID for the moment
```
Comment (antoinebhs) [inline src/components/menus/bus-menu.tsx:113]: will we ever need other info to trip a BBS?

File: src/components/menus/bus-menu.tsx [hunk 2]
```diff
@@ -91,22 +127,45 @@ export const BusMenu: FunctionComponent<BusMenuProps> = ({
...
+                onClick={() => handleTrip()}
                 selected={false}
-                disabled={
-                    computationStarting || oneBusShortcircuitAnalysisState === RunningStatus.RUNNING || !isNodeEditable
-                }
+                disabled={!isNodeEditable}
             >
                 <ListItemIcon>
-                    <BoltIcon />
+                    <OfflineBoltOutlinedIcon />
```
Comment (antoinebhs) [inline src/components/menus/bus-menu.tsx:162]: Can we check with PO that we want the same icon for trip and dynamic SC?

File: src/translations/dynamic/menu-locale-en.ts [hunk 3]
```diff
@@ -16,6 +16,7 @@ const menu_locale_en = {
     Trip2WTransformer: 'Trip the transformer',
     Trip3WTransformer: 'Trip the transformer',
     TripLine: 'Trip the line',
+    TripBusbarSection: 'Trip the busbar section',
```
[the remark this answers is not in the document]
Comment (EstherDarkish) [reply src/translations/dynamic/menu-locale-en.ts:19]: Yes ok

File: src/components/menus/bus-menu.tsx [hunk 4]
```diff
@@ -21,13 +21,19 @@ import { EQUIPMENT_TYPES } from '../utils/equipment-types';
...
+import { tripEquipment } from '../../services/study/network-modifications';
+import { useSnackMessage } from '@gridsuite/commons-ui';
 
 interface BusMenuProps {
     busId: string;
     handleRunShortcircuitAnalysis: (busId: string) => void;
     onOpenDynamicSimulationEventDialog: (equipmentId: string, equipmentType: string, dialogTitle: string) => void;
     position: [number, number];
     onClose: () => void;
+    setModificationInProgress: (arg0: boolean) => void;
```
[the remark this answers is not in the document]
Comment (EstherDarkish) [reply src/components/menus/bus-menu.tsx:36]: Ok, I'll change it.

**Comment by EstherDarkish** [comment]
> * we should disable the option when the BBS is already tripped (similarly to lines or other equipment) to prevent user tripping twice the same BBS
> 
>     * add a "switch on" option to revert a trip ?

Done, the option is now disabled if the busbar section is already tripped.

The "switch on" option is planned for later.

File: src/components/diagrams/singleLineDiagram/single-line-diagram-content.jsx [hunk 5]
```diff
@@ -460,7 +457,7 @@ function SingleLineDiagramContent(props) {
                 isReadyForInteraction ? showEquipmentMenu : null,
 
                 // callback on the buses
-                isReadyForInteraction && shortCircuitAvailability === OptionalServicesStatus.Up ? showBusMenu : null,
```
[the remark this answers is not in the document]
Comment (EstherDarkish) [reply src/components/diagrams/singleLineDiagram/single-line-diagram-content.jsx:463]: Yes, this is a fix

File: src/components/menus/bus-menu.tsx [hunk 6]
```diff
@@ -91,22 +127,45 @@ export const BusMenu: FunctionComponent<BusMenuProps> = ({
...
                 </ListItemIcon>
 
                 <ListItemText
                     primary={
                         <Typography noWrap>
-                            <FormattedMessage id="ShortCircuitAnalysis" />
+                            <FormattedMessage id="TripBusbarSection" />
                         </Typography>
                     }
                 />
```
[the remark this answers is not in the document]
Comment (EstherDarkish) [reply src/components/menus/bus-menu.tsx:187]: Yes, this is a fix

File: src/components/menus/bus-menu.tsx (hunk 2 above)
Comment (EstherDarkish) [reply src/components/menus/bus-menu.tsx:178]: This is the same icon as other trip actions

File: src/components/menus/bus-menu.tsx [hunk 7]
```diff
@@ -44,25 +51,50 @@ const styles = {
...
     busId,
     handleRunShortcircuitAnalysis,
     onOpenDynamicSimulationEventDialog,
     position,
     onClose,
+    setModificationInProgress,
 }) => {
     const [enableDeveloperMode] = useParameterState(PARAM_DEVELOPER_MODE);
+    const { snackError } = useSnackMessage();
+    const [equipmentInfos, setEquipmentInfos] = useState({} as EquipmentInfo);
```
Comment (antoinebhs) [inline src/components/menus/bus-menu.tsx:68]:
```suggestion
    const [equipmentInfos, setEquipmentInfos] = useState<EquipmentInfo>();
```
?

File: src/components/menus/bus-menu.tsx [hunk 8]
```diff
@@ -44,25 +51,50 @@ const styles = {
...
 
+    useEffect(() => {
+        fetchNetworkElementInfos(
+            studyUuid,
+            currentNode?.id,
+            EQUIPMENT_TYPES.BUSBAR_SECTION,
+            EQUIPMENT_INFOS_TYPES.LIST.type,
+            busId,
+            false
+        ).then((value) => {
```
Comment (antoinebhs) [inline src/components/menus/bus-menu.tsx:87]:
```suggestion
        ).then((value: EquipmentInfo | null) => {
```
?

**Review by antoinebhs — APPROVED** [review]
Code OK
Test OK

**Pull request merged**